src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/Items/MsalAccountCacheItem.cs

```diff
@@ -1,65 +1,84 @@
-﻿//----------------------------------------------------------------------
-//
+﻿// ------------------------------------------------------------------------------
+// 
 // Copyright (c) Microsoft Corporation.
 // All rights reserved.
-//
+// 
 // This code is licensed under the MIT License.
-//
+// 
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files(the "Software"), to deal
 // in the Software without restriction, including without limitation the rights
 // to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
 // copies of the Software, and to permit persons to whom the Software is
 // furnished to do so, subject to the following conditions :
-//
+// 
 // The above copyright notice and this permission notice shall be included in
 // all copies or substantial portions of the Software.
-//
+// 
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-//
-//------------------------------------------------------------------------------
+// 
+// ------------------------------------------------------------------------------
 
-using Microsoft.Identity.Core.OAuth2;
 using System.Runtime.Serialization;
+using Microsoft.Identity.Core.Helpers;
+using Microsoft.Identity.Core.OAuth2;
+using Microsoft.Identity.Json.Linq;
 
 namespace Microsoft.Identity.Core.Cache
 {
     [DataContract]
     internal class MsalAccountCacheItem : MsalCacheItemBase
     {
-        internal MsalAccountCacheItem(){
-            AuthorityType = Cache.AuthorityType.MSSTS.ToString();
-        }
-        internal MsalAccountCacheItem(string environment, MsalTokenResponse response) : this()
+        internal MsalAccountCacheItem()
         {
-            IdToken idToken = IdToken.Parse(response.IdToken);
-
-            Init(environment, idToken?.ObjectId, response.ClientInfo, idToken.Name, idToken.PreferredUsername, idToken.TenantId,
-                idToken.GivenName, idToken.FamilyName);
+            AuthorityType = Core.Cache.AuthorityType.MSSTS.ToString();
         }
 
-        internal MsalAccountCacheItem(string environment, MsalTokenResponse response, string preferredUsername, string tenantID) : this()
+        internal MsalAccountCacheItem(
+            string environment,
+            string localAccountId,
+            string rawClientInfo,
+            string name,
```

Comment:
name corresponds to which value?

Reply:
From the spec:


name | Full name for that account including given name and family nameComes from the id_token "name" claim
-- | --




Turns out we don't actually use it in ADAL but it's a primary field in the schema so we're supporting read/write of it.

Reply:
I get that, but could we name it idTokenName or something else instead of just name?

---
In reply to: [255654117](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1498#discussion_r255654117) [](ancestors = 255654117)